Libplanet/Blocks/Block.cs

```diff
@@ -481,33 +490,47 @@ internal void Validate(DateTimeOffset currentTime)
             }
         }
 
-        internal RawBlock ToRawBlock(
-            bool includeHash,
-            bool includeTransactionData
-        )
+        internal BlockDigest ToBlockDigest()
         {
-            IEnumerable<byte[]> transactions =
-                Transactions.OrderBy(tx => tx.Id).Select(
-                    tx => includeTransactionData
-                        ? tx.Serialize(true)
-                        : tx.Id.ToByteArray()
-                );
-            var rawBlock = new RawBlock(
-                index: Index,
-                timestamp: Timestamp.ToString(TimestampFormat, CultureInfo.InvariantCulture),
-                nonce: Nonce.ToByteArray(),
-                miner: Miner?.ToByteArray(),
-                difficulty: Difficulty,
-                transactions: transactions,
-                previousHash: PreviousHash?.ToByteArray()
-            );
+            return new BlockDigest(
+                header: GetBlockHeader(),
+                txIds: Transactions.Select(tx => tx.Id.ToByteArray()));
+        }
+
+        internal RawBlock ToRawBlock()
+        {
+            // For consistency, order transactions by its id.
+            return new RawBlock(
+                header: GetBlockHeader(),
+                transactions: Transactions.OrderBy(tx => tx.Id).Select(tx => tx.Serialize(true)));
+        }
+
+        private byte[] ComputeHash()
```

Comment:
Actually, this method doesn't seem to calculate hash digest yet. 😕

Reply:
Great. I renamed it to `SerializeForHash`.